server/src/main/java/org/apache/iotdb/db/metadata/MTree.java

```diff
@@ -242,23 +242,33 @@ MeasurementMNode createTimeseries(
     // synchronize check and add, we need addChild and add Alias become atomic operation
     // only write on mtree will be synchronized
     synchronized (this) {
-      if (cur.hasChild(leafName)) {
+      MNode child = cur.getChild(leafName);
+      if (child instanceof MeasurementMNode || child instanceof StorageGroupMNode) {
         throw new PathAlreadyExistException(path.getFullPath());
       }
-      if (alias != null && cur.hasChild(alias)) {
-        throw new AliasAlreadyExistException(path.getFullPath(), alias);
+
+      if (alias != null) {
+        MNode childByAlias = cur.getChild(alias);
+        if (childByAlias instanceof MeasurementMNode) {
```

Comment:
how to guarantee childByAlias != null

Reply:
and no UT covers this code segment.

Reply:
> how to guarantee childByAlias != null

if "childByAlias" is null, the statement of "childByAlias instanceof MeasurementMNode" will get "false".